extensions/hibernate-search-orm-elasticsearch/runtime/src/main/java/io/quarkus/hibernate/search/orm/elasticsearch/runtime/management/HibernateSearchPostRequestProcessor.java

```diff
@@ -0,0 +1,214 @@
+package io.quarkus.hibernate.search.orm.elasticsearch.runtime.management;
+
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import org.hibernate.CacheMode;
+import org.hibernate.search.mapper.orm.mapping.SearchMapping;
+import org.hibernate.search.mapper.orm.massindexing.MassIndexer;
+import org.hibernate.search.mapper.orm.scope.SearchScope;
+
+import io.quarkus.arc.Arc;
+import io.quarkus.arc.InstanceHandle;
+import io.quarkus.hibernate.orm.PersistenceUnit;
+import io.vertx.core.buffer.Buffer;
+import io.vertx.core.http.HttpHeaders;
+import io.vertx.core.http.HttpServerRequest;
+import io.vertx.core.http.HttpServerResponse;
+import io.vertx.core.json.JsonArray;
+import io.vertx.core.json.JsonObject;
+import io.vertx.ext.web.RoutingContext;
+
+class HibernateSearchPostRequestProcessor {
+
+    private static final String QUERY_PARAM_BLOCKING = "blocking";
+    private static final String QUERY_PARAM_PERSISTENCE_UNIT = "persistenceUnit";
+    private static final String CONTENT_TYPE_APPLICATION_JSON = "application/json; charset=UTF-8";
+
+    public void process(RoutingContext ctx) {
+        JsonObject config = ctx.body().asJsonObject();
+        if (config == null) {
+            config = new JsonObject();
+        }
+        try (InstanceHandle<SearchMapping> searchMappingInstanceHandle = searchMappingInstanceHandle(ctx.request())) {
+
+            SearchMapping searchMapping = searchMappingInstanceHandle.get();
+
+            JsonObject filter = config.getJsonObject("filter");
+            List<String> types = getTypesToFilter(filter);
+            Set<String> tenants = getTenants(filter);
+            MassIndexer massIndexer;
+            if (types == null || types.isEmpty()) {
+                massIndexer = createMassIndexer(searchMapping.scope(Object.class), tenants);
+            } else {
+                massIndexer = createMassIndexer(searchMapping.scope(Object.class, types), tenants);
+            }
+
+            HibernateSearchMassIndexerConfiguration.configure(
+                    massIndexer, config.getJsonObject("massIndexer"));
+
+            if (isBlockingIndexerRequested(ctx.request())) {
+                try {
+                    massIndexer.startAndWait();
+                } catch (InterruptedException e) {
+                    Thread.currentThread().interrupt();
+                    response(ctx, "error");
+                }
+                response(ctx, "done");
+            } else {
+                massIndexer.start();
+                response(ctx, "started");
```

Comment:
I'm pretty sure blocking here like you do in `massIndexer.startAndWait()` is a big no-no.

When you're asked to block, couldn't you rather pass the future returned by `massIndexer.start()` to the response context, and let Vertx handle the waiting itself?

Also, you might want to have a look at what I did in this demo (I took inspiration from something else in Quarkus, don't remember what): https://github.com/hibernate/hibernate-demos/pull/74/files#diff-e2ecf487e7108639ecb9c840b4fea7cca821fd31f63c48cb91ac85f608bb3765
Basically it implements chunked transfer: when mass indexing starts it sends one chunk with a JSON object aknowledging the request, and when it stops it sends another chunk with a JSON object describing the result.

I'm not saying it absolutely needs to be in this PR, but it might be a good idea to think about how we'd add it, and what we would need to change. I'm thinking that "blocking=true" might not to be expanded to something like `wait_for=started` (response returned when mass indexing started) / `wait_for=progress` (response returns chunked progress) / `wait_for=finished` (response returned when mass indexing finished).
So maybe use `wait_for=started`/`wait_for=finished` for now, so that we can easily add the third option later? Or another name that also fits, I'm just suggesting `wait_for` because Elasticsearch has [something similar](https://www.elastic.co/guide/en/elasticsearch/reference/current/docs-refresh.html).

Two culprits to this "progress" thing though:

* Maybe there's a standard in HTTP to specify whether the request should block or not, should use chunked encoding or not? Some header maybe? I'm really not sure.
* Maybe there's a standard schema the JSON objects for the progress could follow.